### PR TITLE
feat(perf-issues): Add red hatch pattern to affected spans in embedded span tree

### PR DESCRIPTION
**Pull request by 0Calories**
This PR adds a red hatchmark pattern on spans that are part of a performance issue so that they can easily be differentiated from spans that are not directly related to the issue.

### Before
image

### After
image

### To consider in the future
The colours are pretty jarring in dark mode so we'll need to change them later
image

### CODE REVIEW

File: static/app/components/performance/waterfall/constants.tsx [hunk 1]
```diff
@@ -1,2 +1,15 @@
...
+  gap = 'gap',
+  affected = 'affected',
+}
+
+export const SPAN_HATCH_TYPE_COLOURS: Record<
+  SpanBarHatch,
+  {alternate: string; primary: string}
+> = {
+  [SpanBarHatch.gap]: {primary: '#dedae3', alternate: '#f4f2f7'},
+  [SpanBarHatch.affected]: {primary: '#F55459', alternate: '#FAA9AC'},
```
Comment (k-fish) [inline static/app/components/performance/waterfall/constants.tsx:14]: Are these colours net new? Are they part of the theme? If so they should either be pulled from the color consts or we should move these hexs there.

**Review by k-fish — APPROVED** [review]
Code looks fine 👍 , for tests maybe consider adding a data-test-id with the span hatch type now that it's a string and just make sure it gets applied correctly to spans in the span tree.

Aside, but I'm not sure the colours are too much like a 'error' in terms of colors for the affected spans when the main source of the problem (duplicates) are blue. Not suggesting we change anything in this PR since this is an improvement, let's try this for now and see how it looks and feels.

File: static/app/components/performance/waterfall/constants.tsx (hunk 1 above)
Comment (0Calories) [reply static/app/components/performance/waterfall/constants.tsx:14]: I tried passing in the theme obj and getting the colours from there, but our theme doesn't have all of the require colours available. The colours from the gap spans were also hardcoded previously

**Comment by 0Calories** [comment]
I have a fair amount of tests to add in a followup PR, I'll add one for testing this behaviour as well :)

**Pull request merged**